test_projects/cnode/c_src/example/example.spec.exs

```diff
@@ -44,3 +44,10 @@ struct_def(
 )
 
 spec test_my_struct(in_struct :: my_struct) :: {:ok :: label, out_struct :: my_struct}
+
+type outer_struct :: %Outer.Struct{
+  nested_struct: my_struct,
+  id: int
+}
```

Comment:
Maybe
```suggestion
type nested_struct :: %Nested.Struct{
  inner_struct: my_struct,
  id: int
}
```
and then we would have `test_nested_struct(in_struct :: nested_struct)`

`outer_struct` at the first glance is not too intuitive